### PR TITLE
improve stability of test and reducing Duration of Integration Tests Run 

### LINKED ISSUE #242: Extended Duration of Integration Tests Run
**Describe the bug**
The integration-tests run time has been increasing steadily, now taking over 20 minutes. It seems that in some of the integration tests, no images are provided, resulting in the need to destroy the machine due to no OS. This, coupled with a high `gracefulShutdownTimeout`, as seen [here](https://github.com/ironcore-dev/libvirt-provider/blob/615412e3ad6349ed3fa8ec47ffbde4b2d9f35320/provider/server/server_suite_test.go#L34), is causing the extended duration of the integration-tests run.

**Proposed Solution**
To improve the situation, machines should boot properly in integration tests so that the shutdown process can be triggered, ultimately leading to a shorter run duration for the integration tests.

### PR DESCRIPTION
**Pull request by so-sahu**
# Proposed Changes

- Adding squashfsOSImage for machine boot
- Removed the extra fields in machine definitions
- Readjusted the timeouts
- Used ordered flag

Fixes https://github.com/ironcore-dev/libvirt-provider/issues/242

### CODE REVIEW

File: provider/server/status_test.go [hunk 1]
```diff
@@ -37,16 +37,6 @@ var _ = Describe("Status", func() {
 				},
 				Quantity: mcr.GetQuantity(&machineClasses[0], hostResources),
 			},
-			&iriv1alpha1.MachineClassStatus{
```
Comment (lukas016) [inline provider/server/status_test.go:40]: why?

**Review by lukas016 — CHANGES_REQUESTED** [review]

**Comment by lukas016** [comment]
check pls this PR https://github.com/ironcore-dev/libvirt-provider/pull/222/files and move DeferCleanup they are in some files on wrong place and they don't have to be running in some cases.

File: provider/server/server_suite_test.go [hunk 2]
```diff
@@ -78,13 +78,6 @@ var _ = BeforeSuite(func() {
 				MemoryBytes: 8589934592,
 			},
 		},
-		{
```
[the remark this answers is not in the document]
Comment (so-sahu) [reply provider/server/server_suite_test.go:81]: As discussed offline, retaining this configuration can be beneficial for testing and closely resembles the production setup. Therefore, we have decided to keep it.

File: provider/server/status_test.go (hunk 1 above)
Comment (so-sahu) [reply provider/server/status_test.go:40]: As discussed offline, retaining this configuration can be beneficial for testing and closely resembles the production setup. Therefore, we have decided to keep it.

File: provider/server/machine_delete_test.go [hunk 3]
```diff
@@ -77,7 +62,7 @@ var _ = Describe("DeleteMachine", func() {
 		}).Should(Equal(iri.MachineState_MACHINE_RUNNING))
 
 		//allow some time for the vm to boot properly
-		time.Sleep(30 * time.Second)
+		time.Sleep(10 * time.Second)
```
[the remark this answers is not in the document]
Comment (so-sahu) [reply provider/server/machine_delete_test.go]: done

File: provider/server/server_suite_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (so-sahu) [reply provider/server/server_suite_test.go]: reverted

File: provider/server/status_test.go (hunk 1 above)
Comment (so-sahu) [reply provider/server/status_test.go]: reverted

**Comment by lukasfrank** [comment]
@so-sahu  Is there a reason for enforcing the ordered execution?

**Comment by so-sahu** [comment]
I observed some unexpected behavior, particularly when listing machines with labels. Instead of returning one machine, it returned two. It seems that one test is affecting another. Even @lukas016 has observed such strange behavior.

To address this, I'm using ordered execution.

**Comment by lukas016** [comment]
our test aren't written for parallel run

File: provider/server/machine_create_test.go [hunk 4]
```diff
@@ -219,15 +202,10 @@ var _ = Describe("CreateMachine", func() {
...
-				},
+				Metadata: &irimeta.ObjectMetadata{},
 				Spec: &iri.MachineSpec{
-					Power: iri.Power_POWER_ON,
 					Image: &iri.ImageSpec{
-						Image: osImage,
+						Image: squashfsOSImage,
 					},
 					Class:        machineClassx3xlarge,
 					IgnitionData: ignitionData,
```
Comment (lukas016) [inline provider/server/machine_create_test.go:211]: ignition data aren't valid. I recommend delete them because it can impact boot of VM

**Review by lukas016 — CHANGES_REQUESTED** [review]
we are using invalid ignition.

File: provider/server/machine_create_test.go (hunk 4 above)
Comment (so-sahu) [reply provider/server/machine_create_test.go]: removed

**Review by lukas016 — APPROVED** [review]
LGTM

**Comment by so-sahu** [comment]
Re-writing of tests in progress: https://github.com/ironcore-dev/libvirt-provider/pull/345

**Pull request closed without merging**